Add cop to forbid T.untyped in T::Struct props

jeffcarbs opened the pull request:
At Gusto we make heavy use of `T::Struct` and we're converting all of our `ValueObject` (an internal class very similar to T::Struct but without types) to be `T::Struct` instead. During the conversion we're making generous use of `T.untyped` to allow us to safely convert object without breaking anything but we'd like to discourage new `T::Struct`s from having `T.untyped` props.

Morriar commented on spec/cop/sorbet/forbid_untyped_struct_props_spec.rb:
```diff
@@ -0,0 +1,93 @@
+    expect_offense(<<~RUBY)
+      class MyClass < T::Struct
+        const :foo, T.nilable(T.untyped)
+                    ^^^^^^^^^^^^^^^^^^^^ Struct props cannot be T.untyped
```
I would also add a recursive test:

```ruby
class A < T::Struct
  const :foo, T.nilable(T.nilable(T.untyped))
end
```

jeffcarbs replied:
Added a test for nested `T.nilable` and handled that case in the cop.

Morriar commented on spec/cop/sorbet/forbid_untyped_struct_props_spec.rb:
```diff
@@ -0,0 +1,93 @@
+        prop :bar, Date
+        prop :nilable_bar, T.nilable(Float)
+        const :array, T::Array[T.untyped]
+        const :hash, T::Hash[T.untyped, T.untyped]
```
Should it be an option of the cop?

jeffcarbs replied:
Could be a potential future enhancement but I'd prefer to hold off on that within this PR since a lot of edge cases come to mind and I don't have a concrete use-case so I don't know how to decide the right way to implement it (e.g. should `T::Hash[Symbol, T.untyped]` be allowed? What about `T::Hash[Symbol, T::Hash[T.untype, T.untyped]]`).

jeffcarbs commented:
Thanks for the review and sorry about the delay getting back to you. Just added the test you recommended and updated the cop to handle it. Should be ready for another look 👍
